src/ucp/core/ucp_am.c

```diff
@@ -771,14 +840,21 @@ ucp_am_try_send_short(ucp_ep_h ep, uint16_t id, uint32_t flags,
                       const void *header, size_t header_length,
                       const void *buffer, size_t length)
 {
-    if (ucs_likely(!(flags & UCP_AM_SEND_REPLY) &&
-                   ((length == 0) || (header_length == 0)) &&
-                   ((ssize_t)(length + header_length) <=
-                    ucp_ep_config(ep)->am.max_short))) {
+    if (ucs_unlikely(((length != 0) && (header_length != 0)) ||
+                     ((ssize_t)(length + header_length) >
+                      ucp_ep_config(ep)->am.max_short))) {
+        goto out;
+    }
+
+    if (!(flags & UCP_AM_SEND_REPLY)) {
         return ucp_am_send_short(ep, id, flags, header, header_length,
                                  buffer, length);
+    } else if ((length + header_length) < UCP_AM_SHORT_REPLY_MAX_SIZE) {
```

Comment:
can we config ucp_ep_config(ep)->am.max_short to be <= UCP_AM_SHORT_REPLY_MAX_SIZE in the first place?

Reply:
this would shorten `max_short` for non-reply protocol

Reply:
ok, i guess in new proto we can have different max size for am with reply